### ISSUE TITLE
Killing detached process

### ISSUE THREAD

**Issue by joelwurtz** [amphp/process#19]
Actually killing a process which is detached does not work

```php
$process = new \Amp\Process\Process("setsid sleep 100");
$process->start();

$pid = yield $process->getPid();

$process->kill();
```

Then doing a `ps aufx|grep sleep` show the process still running, it is possible to kill him since we have the correct id and we can use then `posix_kill`, but should we do that (it introduce maybe a new extension dependency)

**Comment by kelunik** [comment]
I'm not sure whether it's out job to kill a process which detached. STDIO and everything else still works fine for such a process?

**Comment by joelwurtz** [comment]
Maybe not our job, but some users may not be aware that the process they launch will be detached, so maybe throwing an exception when trying to kill him ?

**Comment by kelunik** [comment]
The thing is that the process the user actually launched is `sh -c`, we already do tricks to get the real processes PID. We should probably use that PID for sending signals, so that'd include `kill()` then.

**Comment by kelunik** [comment]
@bwoebi @trowski Thoughts?

**Comment by Ekstazi** [comment]
Related bug: https://github.com/amphp/process/issues/29

**Comment by trowski** [comment]
I believe this has been resolved by merging #30. If not, please reopen this issue.
